docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,212 @@
+# ML.NET Cookbook
+
+This document is intended to provide essential samples for common usage patterns of ML.NET. 
+It is advisable to be at least minimally familiar with [high-level concepts of ML.NET](MlNetHighLevelConcepts.md), otherwise the terminology in this document may be foreign to you.
+
+## How to use this cookbook
+
+Developers often work by copying and pasting source code from somewhere and then adapting it to their needs. We do it all the time.
+
+So, we decided to embrace the pattern and provide an authoritative set of example usages of ML.NET, for many common scenarios that you may encounter.
+These examples are multi-purpose:
+
+- They can kickstart your development, so that you don't start from nothing,
+- They are annotated and verbose, so you have easier time adapting them to your needs.
+
+Each sample also contains a snippet of the data file used in the sample. We mostly use snippets from our test datasets for that.
+
+Please feel free to search this page and use any code that suits your needs.
+
+### List of recipes
+
+- [How do I load data from a text file?](#how-do-i-load-data-from-a-text-file)
```

Comment:
>(#how-do- [](start = 40, length = 9)

maybe add:
-  one example of reading from more than one file. 
- one example when the schema is completely unknown (default args). First look at the data through TextReader.

Reply:
I don't think I like any of these scenarios too much :)

---
In reply to: [219598238](https://github.com/dotnet/machinelearning/pull/960#discussion_r219598238) [](ancestors = 219598238)